test/middleware.ts

```diff
@@ -1,18 +1,16 @@
-import { Middleware, ParameterizedContext, Response } from 'koa';
-import { AppContext, AppState } from '../src/app';
+import { ExtendableContext } from 'koa';
+import { Middleware } from 'koa-compose';
+import { AppContext } from '../src/app';
 
-export type Next<StateT = AppState, CustomT = AppContext> = (
-  context: ParameterizedContext<StateT, CustomT>
-) => Promise<void>;
+export type Next<CustomT extends ExtendableContext = AppContext> =
+  (context: CustomT) => Promise<void>;
 
-export default async <StateT = AppState, CustomT = AppContext>(
-  middleware: Middleware<StateT, CustomT>,
-  context: ParameterizedContext<StateT, CustomT>,
-  next?: Next<StateT, CustomT>,
-): Promise<Response> => {
+export default async <CustomT extends ExtendableContext>(
+  middleware: Middleware<CustomT>,
+  context: CustomT,
+  next?: Next<CustomT>,
+): Promise<CustomT['response']> => {
```

Comment:
first time I see this notation to navigate to the type of a field, but it's understandable